src/libraries/Microsoft.PowerFx.LanguageServerProtocol/JsonRpc/FormulaTypeJsonConverter.cs

```diff
@@ -60,7 +59,7 @@ public override void Write(Utf8JsonWriter writer, FormulaType value, JsonSeriali
                 DateType => Date,
                 DateTimeType => DateTime,
                 DateTimeNoTimeZoneType => DateTimeNoTimeZone,
-                OptionSetValueType => DateTimeNoTimeZone,
+                OptionSetValueType => OptionSetValue,

```

Comment:
>     OptionSetValueType => OptionSetValue [](http://example.com/codeflow?start=11&length=41)

This is little odd that you can write OptionSetValueType but can't read it back. It gets lost in translation. Is there going to be any issue due to this?

Reply:
Yeah, this is odd. absolutely. Interestingly, it seems like the Read() function here isn't actually used anywhere. I'll track it down with @rsullivan00  and see. There's something a bit tricky about it, since we actually can't create a valid option set value without knowing the backing option set, something we don't have in this scenario.

Reply:
Resolved by #149